Workspace/apps/admin-dashboard/test/functional/adminDashboard.js

```diff
@@ -0,0 +1,216 @@
+                Dash.hamburger().click();
+            });
+            it("clicking on hamburger should toggle treelist navigation mode - collapse and expand", function(){
+                Dash.hamburger()
+                    .visible()
+                    .click()
+                    .click();
+                Dash.treeList()
+                    .and(function(navMenu){
+                        expect(navMenu.getMicro()).toBe(false);
+                    });
+            });
+            describe('click in menu', function(){
+                // Click in menu and check if correct view is loaded
+                it('and navigate to email view', function(){
+                    Dash.menuItem('Email')
+                        .click()
+                        // if selected it should be highlighted with right class
+                        .hasCls('x-treelist-item-selected')
+                        .and(function(menuItem){
+                            expect(menuItem.el.hasCls('x-treelist-item-selected')).toBe(true);
+                        });
+                    Dash.emailView().visible();
+                });
+                it('and navigate to profile view', function(){
+                    Dash.menuItem('Profile')
+                        .click()
+                        .hasCls('x-treelist-item-selected')
+                        .and(function(menuItem){
+                            expect(menuItem.el.hasCls('x-treelist-item-selected')).toBe(true);
+                        });
+                    Dash.profileView().visible();
+                });
+            });
+            describe('navigate in nested menu', function(){
+                beforeEach(function(){
+                    //Need to ensure menu item is collapsed
+                    Dash.menuItem('Pages')
+                        .and(function(menuItem){
+                            menuItem.collapse();
+                        });
+                });
+                it('click to expand menu item', function(){
+                    Dash.menuItem('Pages')
+                        .and(function(menuItem){
+                            expect(menuItem.isExpanded()).toBe(false);
+                        })
+                        .click()
+                        .wait(100) // waiting to make sure proper state will be available
+                        .and(function(menuItem){
+                            expect(menuItem.isExpanded()).toBe(true);
+                        });
+                });
+                describe('click in nested menu', function(){
+                    // Click in nested menu and check if correct view is loaded
+                    beforeEach(function(){
+                        //Need to ensure nested menu items will be visible for testing
+                        Dash.menuItem('Pages')
+                            .and(function(menuItem){
+                                menuItem.expand();
+                            }).wait(1000);
```

Comment:
We need to express what is waited for instead of a number here. The `wait()` method accepts functions that return truthy when the condition is satisfied.